packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -549,20 +551,25 @@ class AndroidSdk {
   }
 
   /// First try Java bundled with Android Studio, then sniff JAVA_HOME, then fallback to PATH.
-  static String findJavaBinary() {
-    if (android_studio.javaPath != null) {
-      return globals.fs.path.join(android_studio.javaPath, 'bin', 'java');
+  static String findJavaBinary({
+    @required AndroidStudio androidStudio,
+    @required FileSystem fs,
```

Comment:
I would use `fileSystem` as the parameter name instead of `fs` so that we don't accidentally confuse it with the global value

Reply:
done